crates/red_knot_python_semantic/src/types.rs

```diff
@@ -613,8 +613,16 @@ impl<'db> Type<'db> {
                 })
             }
 
-            // TODO: handle classes which implement the `__call__` protocol
-            Type::Instance(_instance_ty) => CallOutcome::callable(Type::Todo),
+            Type::Instance(_) => {
+                // Get the class of the instance
+                let meta_ty = self.to_meta_type(db);
```

Comment:
This is just `Type::Class(_instance_ty)`. Should I do that directly, or do we prefer to go through `to_meta_type`?

Reply:
I think here you can just do it directly! Maybe add a comment saying that because it's a dunder it's important to access it as an attribute on the class rather than the instance, though (matching the runtime semantics)